src/main.ts

```diff
@@ -83,41 +121,64 @@ async function main() {
             try {
                 let baseUri = resourceManagerEndpointUrl;
                 if (baseUri.endsWith('/')) {
-                    baseUri = baseUri.substring(0, baseUri.length-1); // need to remove trailing / from resourceManagerEndpointUrl to correctly derive suffixes below
+                    baseUri = baseUri.substring(0, baseUri.length - 1); // need to remove trailing / from resourceManagerEndpointUrl to correctly derive suffixes below
                 }
                 let suffixKeyvault = ".vault" + baseUri.substring(baseUri.indexOf('.')); // keyvault suffix starts with .
-                let suffixStorage = baseUri.substring(baseUri.indexOf('.')+1); // storage suffix starts without .
+                let suffixStorage = baseUri.substring(baseUri.indexOf('.') + 1); // storage suffix starts without .
                 let profileVersion = "2019-03-01-hybrid";
                 await executeAzCliCommand(`cloud register -n "${environment}" --endpoint-resource-manager "${resourceManagerEndpointUrl}" --suffix-keyvault-dns "${suffixKeyvault}" --suffix-storage-endpoint "${suffixStorage}" --profile "${profileVersion}"`, false);
-            } 
+            }
             catch (error) {
                 core.error(`Error while trying to register cloud "${environment}": "${error}"`);
             }
 
             console.log(`Done registering cloud: "${environment}"`)
         }
-    
+
         await executeAzCliCommand(`cloud set -n "${environment}"`, false);
         console.log(`Done setting cloud: "${environment}"`);
 
         // Attempting Az cli login
         if (allowNoSubscriptionsLogin) {
-            let args = [
-                "--allow-no-subscriptions",
-                "--service-principal",
-                "-u", servicePrincipalId,
-                "-p", servicePrincipalKey,
-                "--tenant", tenantId
-            ];
+            var args = [];
+            if (enableOIDC) {
+                args = [
+                    "--allow-no-subscriptions",

```

Comment:
You can create a base args array that contains common keys and based on the specific parameter values you can append to it.
Like if `allowNoSubscriptionsLogin` is true -> append `"--allow-no-subscriptions"` to `args`, same for other keys.

Reply:
Done!✅